.github/workflows/gpuci.yml

```diff
@@ -11,12 +11,25 @@ on:
     # If new commits are added, the "gpuCI" label has to be removed and re-added to rerun gpuCI
     types: [ labeled ]
 
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
```

Comment:
This cancels runs of a PR if a new commit is added to a PR. It does not cancel "foreign runs" (runs of other PRs)